modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_content_organisation\Plugin\ExtraField\Display;
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\extra_field\Plugin\ExtraFieldDisplayFormattedBase;
+use Drupal\oe_content_entity_contact\Entity\ContactInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Display contact information.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_theme_content_organisation_teaser_details",
+ *   label = @Translation("Teaser details"),
+ *   bundles = {
+ *     "node.oe_organisation",
+ *   },
+ *   visible = true
+ * )
+ */
+class TeaserDetailsExtraField extends ExtraFieldDisplayFormattedBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * TeaserDetailsExtraField constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    // Return an empty array if empty, so the field can be considered empty.
+    if ($entity->get('oe_organisation_contact')->isEmpty()) {
+      return [];
+    }
+
+    $contact = $entity->get('oe_organisation_contact')->entity;
+    if (!$contact instanceof ContactInterface) {
+      return [];
+    }
+
+    $build = [
+      '#type' => 'pattern',
+      '#id' => 'field_list',
+      '#variant' => 'horizontal',
+      '#fields' => [
+        'items' => [],
+      ],
+    ];
+    $cache = CacheableMetadata::createFromRenderArray($build);
+
+    $contact_access = $contact->access('view', NULL, TRUE);
+    $cache->addCacheableDependency($contact_access);
+
+    if (!$contact_access->isAllowed()) {
+      $cache->applyTo($build);
+      return $build;
+    }
+
+    $cache->addCacheableDependency($contact);
```

Comment:
Let's maybe move this before line 89? Changes on the entity might have a consequence on its access.